### ISSUE TITLE
ELBO drop for training but not for test set

### ISSUE THREAD

**Issue by xinyixinyijiang** [broadinstitute/CellBender#231]
Hi CellBender group,

Here are the cellbender commands (Linux system) I used for all samples, with default epochs, learning rate and fpr:

```bash
cellbender remove-background \
    --input ${PATH_TO_CELLRANGER_OUTPUT}/${SAMPLE}/outs/raw_feature_bc_matrix.h5 \
    --output ./${SAMPLE}-cb-out/${SAMPLE}-cb-out.h5 \
    --cuda \
    --expected-cells ${EXPECTED_CELLS} \
    --total-droplets-included ${DROPLETS_INCLUDED} \
    --fpr 0.01 \
    --epochs 150
```
However, for sample1, I noticed a drop of ELBO in training set rather than test set. Do you have any idea does this matter? or any possible solution? Many Thanks!
[sample-1-209-cb-out.pdf](https://github.com/broadinstitute/CellBender/files/12093321/sample-1-209-cb-out.pdf)

Here are the QC plots for remaining samples: 
[sample-2-210-cb-out.pdf](https://github.com/broadinstitute/CellBender/files/12093336/sample-2-210-cb-out.pdf)
[sample-3-219-cb-out.pdf](https://github.com/broadinstitute/CellBender/files/12093338/sample-3-219-cb-out.pdf)
[sample-4-220-cb-out.pdf](https://github.com/broadinstitute/CellBender/files/12093339/sample-4-220-cb-out.pdf)
[sample-5-221-cb-out.pdf](https://github.com/broadinstitute/CellBender/files/12093331/sample-5-221-cb-out.pdf)
[sample-6-224-cb-out.pdf](https://github.com/broadinstitute/CellBender/files/12093333/sample-6-224-cb-out.pdf)
[sample-7-225-cb-out.pdf](https://github.com/broadinstitute/CellBender/files/12093334/sample-7-225-cb-out.pdf)
[sample-8-226-cb-out.pdf](https://github.com/broadinstitute/CellBender/files/12093335/sample-8-226-cb-out.pdf)

In sample 2, a concavity is noticeable at around 50 epoch. Do I need to tune parameter for this sample?
In sample3, ELBO decline at around 35 epoch but start to increase at around 45 epoch, I assume this is okay but would be happy if you can confirm it. 

Thank you very much!

**Comment by sjfleming** [comment]
Hi @xinyixinyijiang , sorry for the delay.

Yes, actually I think all these runs look totally fine.  Most look great!  2 and 3 are totally fine... there is always a little wobble in the learning curve, especially early in training.

Even the first sample with a dip in the train ELBO at the end is okay.  The reason this has happened is that cellbender (incorrectly, unfortunately) decided that a couple of the highest-UMI-count droplets (on the left of the plot) are "empty".  This is an error, but there is no way to fix it by tweaking parameters currently.

There will be a new version, v0.3.0, released in the coming days (#238) which will fix this problem.

But for now, if you think those few super-high-UMI count droplets might be doublets anyway, I'd say this run is totally fine.  If you really want those few super-high-UMI count droplets back, then you should re-run using v0.3.0 once it's ready, or install cellbender from the `dev` branch and try now.

**Comment by sjfleming** [comment]
Closed by #238 

Let me know if v0.3.0 does not solve this kind of issue for you!
